### PR TITLE
chore: Add more descriptive error message for auth token parse

### LINKED ISSUE #45: Explore handling auth token parse error more cleanly
Currently if someone copy and pastes an auth token incorrectly into `momento configure` they get the following error
```
 [2022-04-11T21:27:40Z ERROR momento] panicked at 'called `Result::unwrap()` on an `Err` value: ClientSdkError("Failed to parse Auth Token")', client-sdk-rust/src/utils.rs:34:36
```

Can we explore if we can handle this more cleanly and maybe prompt user to double check they copied and pasted their token properly during `configure` command.

### PR DESCRIPTION
**Pull request by poppoerika**
Is users paste an invalid auth token, the following error message will be displayed: `Please make sure entered auth token matches your auth token`

Closes #45

### CODE REVIEW

File: src/jwt.rs [hunk 1]
```diff
@@ -24,9 +24,12 @@ pub fn decode_jwt(jwt: &str) -> Result<Claims, MomentoError> {
     validation.required_spec_claims.insert("cp".to_string());
     validation.validate_exp = false;
     validation.insecure_disable_signature_validation();
-    let token = decode(jwt, &key, &validation)?;
-
-    Ok(token.claims)
+    match decode(jwt, &key, &validation) {
+        Ok(token) => Ok(token.claims),
+        Err(_) => Err(MomentoError::ClientSdkError(
+            "Please make sure entered auth token matches your auth token".to_string(),
```
Comment (bruuuuuuuce) [inline src/jwt.rs:30]: Maybe this should be `Please make sure your auth token is valid` wdyt?

File: src/jwt.rs [hunk 2]
```diff
@@ -44,9 +47,16 @@ mod tests {
...
         let e = decode_jwt("").unwrap_err();
-        let _err_msg = "Failed to parse Auth Token".to_owned();
+        let _err_msg = "Malformed Auth Token".to_owned();
+        assert!(matches!(e, MomentoError::ClientSdkError(_err_msg)));
+    }
+
+    #[test]
+    fn invalid_jwt() {
+        let e = decode_jwt("wfheofhriugheifweif").unwrap_err();
+        let _err_msg = "Please make sure entered auth token matches your auth token".to_owned();
```
Comment (eaddingtonwhite) [inline src/jwt.rs:59]: Maybe reword `Please double check that your auth token was entered correctly` would get @danielamiao buy in here as well.

Comment (danielamiao) [reply src/jwt.rs:59]: `Could not parse token. Please ensure a valid token was entered correctly.`

I just thought it might be helpful to state the problem upfront.

**Review by eaddingtonwhite — APPROVED** [review]

File: src/utils.rs [hunk 3]
```diff
@@ -31,6 +31,11 @@ pub fn is_key_id_valid(key_id: &str) -> Result<(), MomentoError> {
     Ok(())
 }
 
-pub fn get_claims(auth_token: &str) -> Claims {
-    decode_jwt(auth_token).unwrap()
+pub fn get_claims(auth_token: &str) -> Result<Claims, MomentoError> {
+    match decode_jwt(auth_token) {
+        Ok(c) => Ok(c),
+        Err(_) => Err(MomentoError::InvalidArgument(
+            "Cache name cannot be empty".to_string(),
```
[the remark this answers is not in the document]
Comment (wandaitzuchen) [reply src/utils.rs:38]: *returned

File: src/utils.rs [hunk 4]
```diff
@@ -31,6 +31,9 @@ pub fn is_key_id_valid(key_id: &str) -> Result<(), MomentoError> {
     Ok(())
 }
 
-pub fn get_claims(auth_token: &str) -> Claims {
-    decode_jwt(auth_token).unwrap()
+pub fn get_claims(auth_token: &str) -> Result<Claims, MomentoError> {
```
Comment (wandaitzuchen) [inline src/utils.rs:34]: Actually we don't need this function anymore we can use `decode_jwt` directly.

**Review by eaddingtonwhite — APPROVED** [review]
👍 Looking good

**Review by wandaitzuchen — APPROVED** [review]
🚢

**Comment by poppoerika** [comment]
It looks like I'm not authorized to merge this pr...
@eaddingtonwhite - Do you have permission to do so?
image
